modules/templates/proc-adding-templates.adoc

```diff
@@ -0,0 +1,45 @@
+// Module included in the following assemblies:
+//
+// * assemblies/assembly-admin-templates.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="proc-adding-templates_{context}"]
+= Importing an existing template to {product}
+
+You can add an existing template to your {product} instance by using the Catalog Processor.
+
+.Prerequisites
+
+* You have created a directory or repository that contains at least one template YAML file.
+
+.Procedure
+
+. In the `app-config.yaml` configuration file, add the `catalog.configuration` section:
++
+[source,yaml]
+----
+# ...
+catalog:
+  locations:
+    - type: url # <1>
+      target: https://<repository_url>/templates # <2>
+    - type: url
+      target: https://<repository_url>/example-template.yaml
+   - type: file # <3>
+      target: ./templates # <4>
+   - type: file
+      target: ./templates/example-template.yaml
+# ...
+----
+<1> If you are importing templates from a repository, such as GitHub or GitLab, use the `url` type.
+<2> Specify the URL for either a directory where multiple templates are stored, or an individual template as a YAML file.
+<3> If you are importing templates from a local directory, use the `file` type.
+<4> Specify the path to the local directory that contains your template YAML files, or an individual template as a YAML file.
+
+. {product} automatically detects new templates that are added to a configured location, including templates that are added to a configured repository directory, and makes them available in the catalog.
```

Comment:
This goes with the comment above.  I'm not sure that is the case with the static configuration like above.  The only way a new template would get automatically picked up is if you created that location file like mentioned above and that location file added a new template

Reply:
You could do this automatically like you stated if configure the catalog discovery to look for templates.  This is done using a provider but that is specific to each one (GitHub, GitLab).  Here is a link to GitLab's
https://backstage.io/docs/integrations/gitlab/discovery#configuration

Here is a sample configuration:
```
catalog
  rules:
    - allow: [Template]
  providers:
      templates:
        host: yourgitlabhostname
        entityFilename: template.yaml # This assumes your template files all have the same name of template.yaml
        schedule:
          frequency: { minutes: 30 }
          timeout: { minutes: 3 }
```

Reply:
Per the above comment, considering the additional configurations involved, I think that automatically picking up templates will need its own Jira as an advanced use case. It feels like quite a large addition to the scope at this point otherwise for something that was originally scoped as an 8 SP Jira.